src/docker/install.ts

```diff
@@ -33,10 +33,32 @@ import {Exec} from '../exec';
 import {Util} from '../util';
 import {limaYamlData, dockerServiceLogsPs1, setupDockerWinPs1} from './assets';
 import {GitHubRelease} from '../types/github';
+import {HubRepository} from '../hubRepository';
+
+export interface InstallSourceImage {
+  type: 'image';
+  tag: string;
+}
+
+export interface InstallSourceArchive {
+  type: 'archive';
+  version: string;
+  channel: string;
+}
+
+export type InstallSource = InstallSourceImage | InstallSourceArchive;
 
 export interface InstallOpts {
+  source?: InstallSource;
+
+  // @deprecated
+  // Use `source = InstallSourceTypeArchive{version: ..., channel: ...}` instead
   version?: string;
+  // @deprecated
+  // Use `source = InstallSourceTypeArchive{version: ..., channel: ...}` instead
```

Comment:
appreciate the deprecation but this repo is still experimental so you can just remove these fields